nimbus-jose-jwt_aws-kms-extension/src/main/java/com/nimbusds/jose/aws/kms/crypto/impl/KmsAsymmetricSigningCryptoProvider.java

```diff
@@ -86,13 +96,48 @@ public abstract class KmsAsymmetricSigningCryptoProvider extends BaseJWSProvider
+                    .put(JWSAlgorithm.RS384, SigningAlgorithmSpec.RSASSA_PKCS1_V1_5_SHA_384)
+                    .put(JWSAlgorithm.RS512, SigningAlgorithmSpec.RSASSA_PKCS1_V1_5_SHA_512)
+                    .put(JWSAlgorithm.PS256, SigningAlgorithmSpec.RSASSA_PSS_SHA_256)
+                    .put(JWSAlgorithm.PS384, SigningAlgorithmSpec.RSASSA_PSS_SHA_384)
+                    .put(JWSAlgorithm.PS512, SigningAlgorithmSpec.RSASSA_PSS_SHA_512)
+                    .put(JWSAlgorithm.ES256, SigningAlgorithmSpec.ECDSA_SHA_256)
+                    .put(JWSAlgorithm.ES384, SigningAlgorithmSpec.ECDSA_SHA_384)
+                    .put(JWSAlgorithm.ES512, SigningAlgorithmSpec.ECDSA_SHA_512)
+                    // backwards compatibility for KMS-defined algorithm strings
+                    .put(JWSAlgorithm.parse(SigningAlgorithmSpec.RSASSA_PKCS1_V1_5_SHA_256.toString()),
+                            SigningAlgorithmSpec.RSASSA_PKCS1_V1_5_SHA_256)
+                    .put(JWSAlgorithm.parse(SigningAlgorithmSpec.RSASSA_PKCS1_V1_5_SHA_384.toString()),
+                            SigningAlgorithmSpec.RSASSA_PKCS1_V1_5_SHA_384)
+                    .put(JWSAlgorithm.parse(SigningAlgorithmSpec.RSASSA_PKCS1_V1_5_SHA_512.toString()),
+                            SigningAlgorithmSpec.RSASSA_PKCS1_V1_5_SHA_512)
+                    .put(JWSAlgorithm.parse(SigningAlgorithmSpec.RSASSA_PSS_SHA_256.toString()),
+                            SigningAlgorithmSpec.RSASSA_PSS_SHA_256)
+                    .put(JWSAlgorithm.parse(SigningAlgorithmSpec.RSASSA_PSS_SHA_384.toString()),
+                            SigningAlgorithmSpec.RSASSA_PSS_SHA_384)
+                    .put(JWSAlgorithm.parse(SigningAlgorithmSpec.RSASSA_PSS_SHA_512.toString()),
+                            SigningAlgorithmSpec.RSASSA_PSS_SHA_512)
+                    .put(JWSAlgorithm.parse(SigningAlgorithmSpec.ECDSA_SHA_256.toString()),
+                            SigningAlgorithmSpec.ECDSA_SHA_256)
+                    .put(JWSAlgorithm.parse(SigningAlgorithmSpec.ECDSA_SHA_384.toString()),
+                            SigningAlgorithmSpec.ECDSA_SHA_384)
+                    .put(JWSAlgorithm.parse(SigningAlgorithmSpec.ECDSA_SHA_512.toString()),
+                            SigningAlgorithmSpec.ECDSA_SHA_512)
+                    .build();
     /**
      * The supported JWS algorithms (alg).
      * <p>
-     * Note: We are using KMS prescribed algorithm names here.
-     * Ref: https://docs.aws.amazon.com/kms/latest/developerguide/symm-asymm-choose.html#key-spec-rsa-sign
+     * Note: We accept the algorithms defined in RFC-7518 and translate these internally to KMS strings.
+     *
+     * @see <a href="https://datatracker.ietf.org/doc/html/rfc7518#section-3.1">RFC-7518 Section 3.1</a>
+     * @see <a href="https://docs.aws.amazon.com/kms/latest/developerguide/asymmetric-key-specs.html">
+     *     AWS Developer Guide - Asymmetric key specs
+     *     </a>
```

Comment:
This doc-comment also needs to be updated to say, that we support both RFC-7518 algorithm names (defined in JWA spec) and KMS defined algorithm names.

Reply:
👍